registry/src/registry_helper.rs

```diff
@@ -3,17 +3,18 @@
 
 use registry::{Data, Hive, RegKey, Security, key, value};
 use utfx::{U16CString, UCString};
-use crate::config::{Registry, RegistryValueData};
+use crate::config::{Metadata, Registry, RegistryValueData};
 use crate::error::RegistryError;
 
 pub struct RegistryHelper {
     config: Registry,
     hive: Hive,
     subkey: String,
+    what_if: bool,
 }
 
 impl RegistryHelper {
-    pub fn new(config: &str) -> Result<Self, RegistryError> {
+    pub fn new(config: &str, what_if: bool) -> Result<Self, RegistryError> {
```

Comment:
looking at the usage below, I think it would be better to NOT have this as part of the constructor, but instead have it default to `false` and a pub fn `enable_what_if()` that flips to `true`